test/suites/integration/repair_policy_test.go

```diff
@@ -0,0 +1,127 @@
+// Portions Copyright (c) Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration_test
+
+import (
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	karpenterv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	coretest "sigs.k8s.io/karpenter/pkg/test"
+
+	"github.com/Azure/karpenter-provider-azure/test/pkg/environment/common"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/samber/lo"
+)
+
+var _ = Describe("Repair Policy", func() {
+	var selector labels.Selector
+	var dep *appsv1.Deployment
+	var numPods int
+	var unhealthyCondition corev1.NodeCondition
+
+	BeforeEach(func() {
+		unhealthyCondition = corev1.NodeCondition{
+			Type:               corev1.NodeReady,
+			Status:             corev1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: time.Now().Add(-31 * time.Minute)},
+		}
+		numPods = 1
+		// Add pods with a do-not-disrupt annotation so that we can check node metadata before we disrupt
+		dep = coretest.Deployment(coretest.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: coretest.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "my-app",
+					},
+					Annotations: map[string]string{
+						karpenterv1.DoNotDisruptAnnotationKey: "true",
```

Comment:
I'm assuming this is to prevent unexpected consolidation/right-sizing, or such happening during the test?

Probably good to keep here, but does kind of make me wonder if there is a good way for us to test for any thrashing.

Reply:
See comment above, this is to be able to check metadata before disruption